operator/internal/manifests/var.go

```diff
@@ -132,6 +134,22 @@ func serviceAnnotations(serviceName string, enableSigningService bool) map[strin
 	return annotations
 }
 
+func topologySpreadConstraints(spec lokiv1.ReplicationSpec) []corev1.TopologySpreadConstraint {
+	var tsc []corev1.TopologySpreadConstraint
+	if len(spec.Zones) > 0 {
+		tsc = make([]corev1.TopologySpreadConstraint, len(spec.Zones))
+		for i, z := range spec.Zones {
+			tsc[i] = corev1.TopologySpreadConstraint{
+				MaxSkew:           int32(z.MaxSkew),
+				TopologyKey:       z.TopologyKey,
+				WhenUnsatisfiable: corev1.DoNotSchedule,
+			}
+		}
+	}
```

Comment:
I don't understand your suggestion, as I understand it, it will result in a panic, unless I missed something here.
The check is there in order to return nil when no zones are defined instead of an empty slice.

Reply:
Sorry my fault I forgot to pull the make in there:

Reply:
```suggestion
	tsc := make([]corev1.TopologySpreadConstraint, 0)
	for i, z := range spec.Zones {
		tsc[i] = corev1.TopologySpreadConstraint{
			MaxSkew:           int32(z.MaxSkew),
			TopologyKey:       z.TopologyKey,
			WhenUnsatisfiable: corev1.DoNotSchedule,
		}
	}

```

Reply:
>  The check is there in order to return nil when no zones are defined instead of an empty slice.